javascript/extractor/src/com/semmle/js/parser/TypeScriptASTConverter.java

```diff
@@ -2179,9 +2180,24 @@ private Node convertTryStatement(JsonObject node, SourceLocation loc) throws Par
   }
 
   private Node convertTupleType(JsonObject node, SourceLocation loc) throws ParseError {
-    return new TupleTypeExpr(loc, convertChildrenAsTypes(node, "elements"));
+    List<JsonElement> elements = new ArrayList<>();
+    ((JsonArray)node.get("elements")).iterator().forEachRemaining(elements::add);
+
+    List<String> elementNames = elements.stream()
+      .filter(n -> getKind(n).equals("NamedTupleMember"))
+      .map(n -> n.getAsJsonObject().get("name"))
+      .map(n -> n.getAsJsonObject().get("escapedText"))
+      .map(n -> n.getAsString())
+      .collect(Collectors.toList());
```

Comment:
We should extract an AST node for the identifier instead of just the name.

(Also, we have `getIdentifierText` instead of reading `"escapedText"` manually, but this will likely not matter if extracting the identifier)